base/error.jl

```diff
@@ -52,28 +52,31 @@ end
 
 
 """
-    retry(f, [condition]; n=3; max_delay=10) -> Function
+    retry(f, [on_error]; n=1; max_delay=0) -> Function
```

Comment:
What is the objection to naming the optional retry condition `condition` ?
`on_error` sounds like an action to take when an error occurs.